forte/orbital-helpers/orbital_embedding.cc

```diff
@@ -906,4 +1005,192 @@ psi::SharedMatrix semicanonicalize_block(psi::SharedWavefunction ref_wfn, psi::S
         return C_block;
     }
 }
+
+// Utility function #2: CAS-AO Fock builder
+void Build_CAS_AO_Fock(psi::SharedWavefunction ref_wfn, int nirrep, psi::Dimension doccpi,

```

Comment:
This seems a repetition of the Fock build of the integral class. I would suggest to use what's available in the integral class for two reasons: 1) current format does not work for FCIDUMP; 2) code maintenance. I think the embedding is done before integrals initialized. If so, I would temporarily add some bypass for using Psi4 integrals.

Reply:
I am still confused. When I am doing this embedding step, there is no ForteIntegral class exist. So is it means I need to build a virtual ForteIntegral, put in the value, and run the Fock build functions? This seems to be very circumventing.

Reply:
Here we only need the Fock build from JK, which are available in ForteIntegral. I am not sure it is a good practice because Francesco mention this to me for MCSCF bypassing the integral transformation in ForteIntegral. But what I did for MCSCF is just initialize the JK in ForteIntegral but not store all the integrals. So in principle you could do the same thing.